2.cellprofiling/featurization_utils/area_size_shape_utils.py

```diff
@@ -0,0 +1,118 @@
+import numpy
+import skimage.measure
+from loading_classes import ImageSetLoader, ObjectLoader
+
+
+def calulate_surface_area(
+    label_object: numpy.array,
+    props: numpy.array,
+    spacing: tuple,
+) -> list:
+    """
+    This function calculates the surface area of each object in a 3D image using the marching cubes algorithm.
+    Parameters
+    ----------
+    label_object : numpy.array
+        This is an array of the segmented objects of a given compartment.
+    props : numpy.array
+        This is the output of the regionprops function, which contains information about the objects.
+    spacing : tuple
+        This is the spacing of the image in each dimension (z, y, x).
+
+    Returns
+    -------
+    list
+        A list of surface areas for each object in the image.
+    """
+
+    # this seems less elegant than you might wish, given that regionprops returns a slice,
+    # but we need to expand the slice out by one voxel in each direction, or surface area freaks out
+    surface_areas = []
+    for index, label in enumerate(props["label"]):
+
+        volume = label_object[
+            max(props["bbox-0"][index] - 1, 0) : min(
+                props["bbox-3"][index] + 1, label_object.shape[0]
+            ),
+            max(props["bbox-1"][index] - 1, 0) : min(
+                props["bbox-4"][index] + 1, label_object.shape[1]
+            ),
+            max(props["bbox-2"][index] - 1, 0) : min(
+                props["bbox-5"][index] + 1, label_object.shape[2]
+            ),
```

Comment:
Just to confirm, because it is in 3D that is why there are six bbox-es?

This is what I could find on it so just looking for clarification :)
```
Bounding box (min_row, min_col, max_row, max_col)
```

Reply:
Nevermind I got my answer below lol